logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -12,7 +12,8 @@ def initialize(logger, output_class, metric, strategy_registry, plugin_args)
     @output_class = output_class
     @metric = metric
     @plugin_args = plugin_args
-    @strategy_registry ||= ::LogStash::OutputDelegatorStrategyRegistry.instance
+    @strategy_registry = strategy_registry
```

Comment:
This is a problem because it is not clear in the next line whether your intention is to test for missing arg `strategy_registry` or for missing setting via the attr_reader.

Consider changing the arg name to perhaps `strat_registry` or using the ivar in line 16 (but then line 17 becomes weird e.g. ivar in L16 and attr_reader in L17)